test/concurrent/ThreadTest.ooc

```diff
@@ -45,15 +45,17 @@ ThreadTest: class extends Fixture {
 				value set(value get() + 1)
 				Thread yield()
 				Time sleepMilli(1)
+				if (Thread currentThread() isCanceled())
```

Comment:
@sebastianbaginski It's not too complex to implement, but it requires more effort/complexity to actually get something cancelable running on a thread. I prefered my approach because it would allow you to just supply a func the same way as before, but I see now we will need a global `ThreadTable` to handle it, and the beauty goes away.

I guess we can keep the `Func` constructor for `Thread` that automatically creates a (non-cancelable) task and disposes it when the thread is freed, that way we don't need to change as much in the code. Cancelable tasks would need to be manually created and supplied to the thread. We already have a `Task` system for the `ThreadPool` that may or may be partially replaced by this new `ThreadTask`.

I guess I will close this for now and we can pick up the discussion in #1588 or IRL.